pkgs/development/libraries/ffmpeg/default.nix

```diff
@@ -0,0 +1,58 @@
+{ callPackage, darwin }:
+
+let
+  mkFFmpeg =
+    initArgs: ffmpegVariant:
+    callPackage ./generic.nix (
+      {
+        inherit (darwin.apple_sdk.frameworks)
+          Cocoa
+          CoreServices
+          CoreAudio
+          CoreMedia
+          AVFoundation
+          MediaToolbox
+          VideoDecodeAcceleration
+          VideoToolbox
+          ;
```

Comment:
This can be moved to the generic file.

Like I did in f5f3d90e4fef6413e11fbe323ef1925ac634f11c from #292498

Reply:
Your work there is also good but I kinda prefer this honestly.

Reason being that we get control over individual deps in the override interface and they're still defined within the package; best of both worlds.

Reply:
Your explanation makes sense to me.